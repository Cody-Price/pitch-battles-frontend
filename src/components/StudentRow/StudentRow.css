.class-row {
  position: relative;
  overflow: hidden;
}

.student-select-clicker:hover {
  cursor: pointer;
}

.student-row-delete:hover {
  cursor: pointer;
}

.student-delete-warning {
  background: #d3d3d3;
  display: flex;
  position: absolute;
  width: 450px;
  justify-content: space-around;
  align-items: center;
  box-shadow: 2px 2px 2px rgba(0, 0, 0, 0.3);
  right: -600px;
  transition: 0.3s;
}

.student-delete-warning.true {
  right: 0;
}
