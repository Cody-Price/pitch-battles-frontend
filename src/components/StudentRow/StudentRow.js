import React, { Component } from "react";
import "./StudentRow.css";

class StudentRow extends Component {
  render() {
    const { student, selectStudent } = this.props;
    return (
      <article className="class-row">
        <p onClick={() => selectStudent(student)}>
          {student.attributes.first_name} {student.attributes.last_name}
        </p>
        <p>{student.attributes.total_games_played}</p>
        <p>{student.attributes.level_one_fastest_time / 1000}</p>
        <p>{student.attributes.level_two_fastest_time / 1000}</p>
        <p>{student.attributes.level_three_fastest_time / 1000}</p>
        <p>{student.attributes.level_four_fastest_time / 1000}</p>
        <p>Delete</p>
      </article>
    );
  }
}

export default StudentRow;
