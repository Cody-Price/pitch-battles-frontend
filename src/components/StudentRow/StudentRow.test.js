import { shallow } from "enzyme";
import React from "react";
import StudentRow from "./StudentRow";
const Enzyme = require("enzyme");
const EnzymeAdapter = require("enzyme-adapter-react-16");

Enzyme.configure({ adapter: new EnzymeAdapter() });

jest.mock("../../utilities/fetchCalls");

describe("StudentRow", () => {
  let wrapper;

  const mockStudent = {
    attributes: {
      first_name: "kevin",
      last_name: "simpson",
      level_one_fastest_time: 1000,
      level_two_fastest_time: 2000,
      level_three_fastest_time: 3000,
      level_four_fastest_time: 4000
    }
  };
  const mockSelectStudent = jest.fn();

  beforeEach(() => {
    wrapper = shallow(
      <StudentRow student={mockStudent} selectStudent={mockSelectStudent} />
    );
  });

  it("should match the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should call selectStudent on click", () => {
    wrapper.find(".student-select-clicker").simulate("click");

    expect(mockSelectStudent).toHaveBeenCalledWith(mockStudent);
  });
});
