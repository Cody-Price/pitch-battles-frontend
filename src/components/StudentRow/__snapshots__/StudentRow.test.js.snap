// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StudentRow should match the snapshot 1`] = `
<article
  className="class-row"
>
  <p
    className="student-select-clicker"
    onClick={[Function]}
  >
    kevin
     
    simpson
  </p>
  <p />
  <p>
    1
  </p>
  <p>
    2
  </p>
  <p>
    3
  </p>
  <p>
    4
  </p>
  <p
    className="student-row-delete"
    onClick={[Function]}
  >
    Delete
  </p>
  <div
    className="student-delete-warning false"
  >
    <p>
      Remove 
      kevin
       from the class?
    </p>
    <button
      className="delete-student-btn"
      onClick={[Function]}
    >
      remove
    </button>
    <button
      className="cancel-student-delete"
      onClick={[Function]}
    >
      cancel
    </button>
  </div>
</article>
`;
